Sming/SmingCore/Data/HttpHeaders.h

```diff
@@ -0,0 +1,147 @@
+/****
+ * Sming Framework Project - Open Source framework for high efficiency native ESP8266 development.
+ * Created 2015 by Skurydin Alexey
+ * http://github.com/SmingHub/Sming
+ * All files of the Sming Core are provided under the LGPL v3 license.
+ *
+ * @author: 2018 - Mikee47 <mike@sillyhouse.net>
+ *
+ *  Encapsulate encoding and decoding of HTTP header fields
+ *  Used for HTTP connections and SMTP mail
+ *
+ *  The HttpHeaders class was an empty HashMap class living in 'Structures.h'.
+ *  It has been expanded here to simplify code, and to provide enumerated keys
+ *  for common field names.
+ *
+ ****/
+
+#ifndef _SMING_CORE_DATA_HTTP_HEADERS_H_
+#define _SMING_CORE_DATA_HTTP_HEADERS_H_
+
+#include "WString.h"
+#include "WHashMap.h"
+
+/*
+ * Common HTTP header field names. Enumerating these simplifies matching
+ * and generating headers. The strings themselves are stored in flash to
+ * save RAM.
+ *
+ * According to RFC 2616: 4.2, field names are case-insensitive.
+ *
+ */
+#define HTTP_HEADER_FIELDNAME_MAP(XX)                                                                                  \
+	XX(AccessControlAllowOrigin, "Access-Control-Allow-Origin", "")                                                    \

```

Comment:
Replace the first element to be upper-case words separated with underscore. This way it will be quite natural to set the header to `HTTP_HEADER_AUTHORIZATION` or `HTTP_HEADER_ACCESS_CONTROL_ALLOW_ORIGIN`.